src/rosdep2/sources_list.py

```diff
@@ -501,16 +537,23 @@ def update_sources_list(sources_list_dir=None, sources_cache_dir=None,
         raise ValueError(
             'Requested distribution "%s" is not in the index.' % ros_distro)
 
-    for dist_name in sorted(distribution_names):
-        distribution = get_index().distributions[dist_name]
+    dist_names = retry(max_retries,
+                       lambda: sorted(get_index().distributions.keys()),
+                       (DownloadFailure, IOError))
+
+    """
+    This closure uses variables from parent scope
+    """
+    def get_additional_sources(dist_name):
         if dist_name != ros_distro:
             if ros_distro is not None:
                 print('Skip distro "%s" different from requested "%s"' % (dist_name, ros_distro))
-                continue
-            if skip_eol_distros:
-                if distribution.get('distribution_status') == 'end-of-life':
-                    print('Skip end-of-life distro "%s"' % dist_name)
-                    continue
+                return
+        distribution = get_index().distributions[dist_name]
+        if skip_eol_distros:
+            if distribution.get('distribution_status') == 'end-of-life':
+                print('Skip end-of-life distro "%s"' % dist_name)
+                return
```

Comment:
This is indented 1 level in the original commit, but I felt the indentation was wrong. PTAL, I can revert this if needed